### PR TITLE
[MLIR][Presburger] Make printing aligned to assist in debugging

### PR DESCRIPTION
**Pull request by kaitingwang**
Hello Arjun! Please allow me to contribute this patch as it helps me debugging significantly!  When the 1's and 0's don't line up when debugging farkas lemma of numerous polyhedrons using simplex lexmin solver, it is truly straining on the eyes.  Hopefully this patch can help others!  

The unfortunate part is the lack of testcase as I'm not sure how to add testcase for debug dumps.  :)   However, you can add this testcase to the SimplexTest.cpp to witness the nice printing!

```c++
TEST(SimplexTest, DumpTest) {
  int COLUMNS = 2;
  int ROWS = 2;
  LexSimplex simplex(COLUMNS * 2);
  IntMatrix m1(ROWS, COLUMNS * 2 + 1);
  // Adding LHS columns.
  for (int i = 0; i < ROWS; i++) {
    // an arbitrary formula to test all kinds of integers
    for (int j = 0; j < COLUMNS; j++) 
      m1(i, j) = i + (2 << (i % 3)) * (-1 * ((i + j) % 2));
  }
  // Adding RHS columns.
  for (int i = 0; i < ROWS; i++) {
    for (int j = 0; j < COLUMNS; j++)
      m1(i, j + COLUMNS) = j - (3 << (j % 4)) * (-1 * ((i + j * 2) % 2));
  }
  for (int i = 0; i < m1.getNumRows(); i++) {
    ArrayRef<DynamicAPInt> curRow = m1.getRow(i);
    simplex.addInequality(curRow);
  }
  IntegerRelation rel =
      parseRelationFromSet("(x, y, z)[] : (z - x - 17 * y == 0, x - 11 * z >= 1)",2);
  simplex.dump();
  m1.dump();
  rel.dump();
}
```

```
rows = 2, columns = 7
var: c3, c4, c5, c6
con: r0 [>=0], r1 [>=0]
r0: -1, r1: -2
c0: denom, c1: const, c2: 2147483647, c3: 0, c4: 1, c5: 2, c6: 3
  1  0  1  0 -2  0  1
  1  0 -8 -3  1  3  7

  0 -2  0  1  0
 -3  1  3  7  0
Domain: 2, Range: 1, Symbols: 0, Locals: 0
2 constraints
 -1  -17  1   0   = 0
  1   0  -11 -1  >= 0

```

### CODE REVIEW

File: mlir/include/mlir/Analysis/Presburger/Utils.h
Comment (Groverkss) [inline mlir/include/mlir/Analysis/Presburger/Utils.h:302]: can you write out an example of what this is here? I can't tell from the current description.

File: mlir/include/mlir/Analysis/Presburger/Utils.h
Comment (Groverkss) [inline mlir/include/mlir/Analysis/Presburger/Utils.h:307]: nit: use /// for function descriptions instead of //

File: mlir/include/mlir/Analysis/Presburger/Utils.h
Comment (Groverkss) [inline mlir/include/mlir/Analysis/Presburger/Utils.h:312]: nit: str.empty()

File: mlir/include/mlir/Analysis/Presburger/Utils.h
Comment (Groverkss) [inline mlir/include/mlir/Analysis/Presburger/Utils.h:314]: nit: unsigned int -> unsigned

File: mlir/include/mlir/Analysis/Presburger/Utils.h
Comment (Groverkss) [inline mlir/include/mlir/Analysis/Presburger/Utils.h:321]: nit: /// (same as previous comment)

File: mlir/include/mlir/Analysis/Presburger/Utils.h
Comment (Groverkss) [inline mlir/include/mlir/Analysis/Presburger/Utils.h:333]: nit: use braces for else

**Review by Groverkss — APPROVED** [review]
LGTM, Thanks!

**Comment by kaitingwang** [comment]
```
Assume we'd like to print the following three numbers:
.12
3.4
56.7

 prealign = "."
minSpacing = 1

 First we iterate over all vals and compute the maxPreIndent and maxPostIndent:

String  PreIndent              PostIndent          Actual PreIndent              Actual PostIndent

.12     1                       2                        3 - (1 - 1) = 3              2 - (2) = 0

3.4     2                       1                        3 - (2 - 1) = 2              2 - (1) = 1

56.7    3                       1                        3 - (3 - 1) = 1              2 - (1) = 1
------------------------------------------------------------
max     3                       2

 
Then we perform the actual print where number of spaces printed are the Actual PreIndent and PostIndent caculated above.
 

Actual_PreIndent = maxPreIndent - (preIndent - minSpacing)
Actual_PosIndent = maxPostIndent - (postIndent)

 To illustrate:
` = Actual_PreIndent
~ = Actual_PostIndent

```.12```.12
``3.4~``3.4~
`56.7~`56.7~
```

File: mlir/include/mlir/Analysis/Presburger/Utils.h
Comment (kaitingwang) [reply mlir/include/mlir/Analysis/Presburger/Utils.h]: Fixed with an example.

File: mlir/include/mlir/Analysis/Presburger/Utils.h
Comment (kaitingwang) [reply mlir/include/mlir/Analysis/Presburger/Utils.h]: Fixed as suggested.

File: mlir/include/mlir/Analysis/Presburger/Utils.h
Comment (kaitingwang) [reply mlir/include/mlir/Analysis/Presburger/Utils.h]: Fixed as suggested.

File: mlir/include/mlir/Analysis/Presburger/Utils.h
Comment (kaitingwang) [reply mlir/include/mlir/Analysis/Presburger/Utils.h]: Fixed as suggested.

File: mlir/include/mlir/Analysis/Presburger/Utils.h
Comment (kaitingwang) [reply mlir/include/mlir/Analysis/Presburger/Utils.h]: Fixed as suggested.

File: mlir/include/mlir/Analysis/Presburger/Utils.h
Comment (kaitingwang) [reply mlir/include/mlir/Analysis/Presburger/Utils.h]: Fixed as suggested.

**Comment by kaitingwang** [comment]
@Groverkss Thanks for the review! All comments addressed.

File: mlir/include/mlir/Analysis/Presburger/Utils.h
Comment (Groverkss) [inline mlir/include/mlir/Analysis/Presburger/Utils.h:297]:
```suggestion
/// Example usage:
/// Print .12, 3.4, 56.7 
/// preAlign = ".", minSpacing = 1,
```

File: mlir/include/mlir/Analysis/Presburger/Utils.h
Comment (Groverkss) [inline mlir/include/mlir/Analysis/Presburger/Utils.h:298]: Its fine to just put spaces and not have special characters here.

File: mlir/include/mlir/Analysis/Presburger/Utils.h
Comment (Groverkss) [inline mlir/include/mlir/Analysis/Presburger/Utils.h:332]: nit: !str.empty()

File: mlir/lib/Analysis/Presburger/IntegerRelation.cpp
Comment (Groverkss) [inline mlir/lib/Analysis/Presburger/IntegerRelation.cpp:2598]: nit: use braces if outer loop is using braces

File: mlir/lib/Analysis/Presburger/IntegerRelation.cpp
Comment (Groverkss) [inline mlir/lib/Analysis/Presburger/IntegerRelation.cpp:2602]: Same as above

**Review by Groverkss — APPROVED** [review]
Thank you for the patch! I added some style preferences as comments. Feel free to land them once addressed without waiting for me (or ping me if you need me to land this)

File: mlir/include/mlir/Analysis/Presburger/Utils.h
Comment (kaitingwang) [reply mlir/include/mlir/Analysis/Presburger/Utils.h]: Updated.

File: mlir/include/mlir/Analysis/Presburger/Utils.h
Comment (kaitingwang) [reply mlir/include/mlir/Analysis/Presburger/Utils.h]: Removed special characters and used space.

File: mlir/include/mlir/Analysis/Presburger/Utils.h
Comment (kaitingwang) [reply mlir/include/mlir/Analysis/Presburger/Utils.h]: Fixed.

File: mlir/lib/Analysis/Presburger/IntegerRelation.cpp
Comment (kaitingwang) [reply mlir/lib/Analysis/Presburger/IntegerRelation.cpp:2598]: This is a good point!  Not sure how I never noticed this rule.  Thank you for pointing it out.

File: mlir/lib/Analysis/Presburger/IntegerRelation.cpp
Comment (kaitingwang) [reply mlir/lib/Analysis/Presburger/IntegerRelation.cpp:2601]: Fixed.

**Comment by kaitingwang** [comment]
@Groverkss Thanks for the comments! (I've learned a thing or two)  All fixes are in.  I'll wait until CI turns green before landing the patch.

**Pull request merged**
